packages/cozy-clisk/src/contentscript/ContentScript.js

```diff
@@ -268,6 +299,44 @@ export default class ContentScript {
     return true
   }
 
+  /**
+   * Wait for the given labelled request to be intercepted. The labeled request must be defined and
+   * sent to the ContentScript constructor
+   *
+   * @param {string} label - any label string defined in the RequestInterceptor
+   * @param {object} [options] - options object
+   * @param {number} [options.timeout] - number of miliseconds before the function sends a timeout error. Default 60000ms
+   */
+  waitForRequestInterception(label, options = {}) {
+    this.onlyIn(PILOT_TYPE, 'waitForRequestInterception')
+    const timeout = options?.timeout ?? 60000
+
+    const interceptionPromise = new Promise(resolve => {
+      const listener = ({ event, payload }) => {
+        if (event === 'requestResponse' && payload.label === label) {
+          if (!this.bridge) {
+            throw new Error(
+              'No bridge is defined, you should call ContentScript.init before using this method'
+            )
+          }
```

Comment:
no need to add this test, you already added it before adding the listener

Reply:
This was actually ts-check which enforces me to do it. In theory, nothing avoid to remove the bridge between the two tests :shrug: